deploy/docker/Dockerfile

```diff
@@ -1,13 +1,19 @@
 FROM golang:1.15.1 AS builder
 
+# These vars are not persisted in the final image layer
+ARG KUIPER_USER="kuiper"
+ARG KUIPER_USER_ID="1001"
+
 COPY . /go/kuiper
 
 WORKDIR /go/kuiper
 
-RUN apt update && apt install -y pkg-config libczmq-dev
+RUN apt update && apt install -y pkg-config libczmq-dev && \
+    adduser -q --no-create-home --uid ${KUIPER_USER_ID} --disabled-password --shell /usr/sbin/nologin --gecos "" ${KUIPER_USER}
 
 RUN make build_with_edgex \
-    && ln -s /go/kuiper/_build/kuiper-$(git describe --tags --always)-$(uname -s | tr "[A-Z]" "[a-z]")-$(uname -m) /usr/local/kuiper 
+    && ln -s /go/kuiper/_build/kuiper-$(git describe --tags --always)-$(uname -s | tr "[A-Z]" "[a-z]")-$(uname -m) /usr/local/kuiper && \
+    chown -Rh ${KUIPER_USER}:${KUIPER_USER} /go/kuiper/_build/kuiper-$(git describe --tags --always)-$(uname -s | tr "[A-Z]" "[a-z]")-$(uname -m)
 
```

Comment:
I think `adduser` and `chown` can be combined in the same `RUN`, which makes it easier to read, don't you think?

Reply:
No, they are separated because of build timing - "root" has to create the user and perform build processes, and then assign ownership after the build processes are complete.

Reply:
In this instance, I have it separated because of the `make build_with_edgex` and the link that it's creating. The ownership lies in the build output directory that `/usr/local/kuiper` is linked to post build. I figure I'd run it after build so that it can build with root and assume ownership after it's all done.